cmd/registry-replacer/main.go

```diff
@@ -278,6 +297,19 @@ func replacer(
 			if err := pruneUnusedReplacements(config, allReplacementCandidates); err != nil {
 				return fmt.Errorf("failed to prune unused replacements: %w", err)
 			}
+
+			// to prune base images we'll need the fully-resolved config.
+			resolvedConfig, err := configResolver()
+			if err != nil {
+				// TODO this should stop execution rather than just logging an error, but since we want a smooth transition with
+				// the new registry-replacer enhancements we'll add that in later, after this has been rolled out.
+				logrus.Error(fmt.Errorf("failed to resolve config. This means that base image pruning will not function as expected and will therefore be skipped: %w", err))
+			} else {
+				err := pruneUnusedBaseImages(config, resolvedConfig)
```

Comment:
Nit: Doing it like below is more concise and reduces the scope of the `err` variable (although the latter is irrelevant here):
```suggestion
				if err := pruneUnusedBaseImages(config, resolvedConfig); err != nil {
				   handle
				 }
```

Reply:
I keep forgetting that go has this, definitely nicer!